### PR TITLE
Map creation

### LINKED ISSUE #304: Map creation
Let users create and delete maps.
Implements [map creation and deletion use case.](https://github.com/ElektraInitiative/PermaplanT/blob/df25c5691321c7f324302fd35994e5dc8f4d71e8/doc/usecases/map_create_delete.md)

- [x] Overview page for the maps of an user.
- [x] Backend API for adding a new map to the database.
- [x] Frontend form for creating a new map.

### PR DESCRIPTION
**Pull request by ThorbenStaufer**
Adding a map overview page to the frontend, showing a list of created maps. 
New maps can be created through a form reachable via the map overview.
A map can also be set to inactive, permanently removing it in 30 days unless it is reactivated.

_This PR is currently a draft and not all features are implemented_

**Current Progress:**

- [x] Overview page for the maps of an user.
- [x] Backend API for adding a new map to the database.
- [x] Frontend form for creating a new map.
- [ ] Add tests for backend.

**ToDos for future PRs:**

- [ ] Adding Geo Data to maps.
- [ ] Set the owner of the map to the user that created it.
- [ ] Fetching only owned maps in the map overview.
- [ ] Moving map overview to a profile page.

Will resolve #304.

## Basics

- [ ] Details of what you changed are in commit messages.
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildserver is happy.
- [ ] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed all affected decisions
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Documentation is conforming to [our Documentation Guidelines](/doc/documentation.md)
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to our Coding Guidelines
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

### CODE REVIEW

File: backend/src/model/dto.rs [hunk 1]
```diff
@@ -175,3 +180,73 @@ pub struct Page<T> {
     /// Number of pages in total.
     pub total_pages: i32,
 }
+
+#[allow(clippy::missing_docs_in_private_items)] // TODO: See #97.
```
Comment (kitzbergerg) [inline backend/src/model/dto.rs:184]: Please document the structs and their fields. #97 is already close, so it should be referenced anywhere anymore.

File: backend/src/service/map.rs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+/// Save a snapshot of the map as a new version.
+///
+/// # Errors
+/// If the connection to the database could not be established.
+pub async fn save_snapshot(
+    new_map_version: NewMapVersionDto,
+    pool: &Data<Pool>,
+) -> Result<MapVersionDto, ServiceError> {
+    // TODO: create element entries for all layers with reference to this version.
```
Comment (kitzbergerg) [inline backend/src/service/map.rs:75]: If you don't plan to resolve this TODO in this PR maybe create a new issue an reference it?

**Review by kitzbergerg — CHANGES_REQUESTED** [review]
Looks good. I haven't looked at the code for the frontend.

Please also create tests in the `test/` folder. Make a new file for maps and create at least one test for each endpoint. If there is more complicated logic maybe more and even negative tests (meaning the server return status code 4xx)

File: backend/src/service/map.rs (hunk 2 above)
Comment (ThorbenStaufer) [reply backend/src/service/map.rs:75]: Issue #341 now includes a description of this future TODO.

File: backend/src/controller/map.rs [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        service::map::find(search_query.into_inner(), page_query.into_inner(), &pool).await?;
+    Ok(HttpResponse::Ok().json(response))
+}
+
+/// Endpoint for fetching a [`Map`](crate::model::entity::Map).
+///
+/// # Errors
+/// * If the connection to the database could not be established.
+#[utoipa::path(
+    context_path = "/api/users/{user_id}/maps/{map_id}",
```
Comment (kitzbergerg) [inline backend/src/controller/map.rs:47]: Once auth works we might have to revise that (to either take the user_id directly from the token or at least validate it).
There is nothing to change here just something to remember.

File: backend/src/controller/map.rs [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    responses(
+        (status = 201, description = "Save a new map version", body = MapVersionDto)
+    )
+)]
+#[post("/{map_id}/versions")]
+pub async fn save_snapshot(
+    new_map_version_json: Json<NewMapVersionDto>,
+    pool: Data<Pool>,
+) -> Result<HttpResponse> {
+    let response = service::map::save_snapshot(new_map_version_json.0, &pool).await?;
```
Comment (kitzbergerg) [inline backend/src/controller/map.rs:119]: Same thing with auth here. Probably need a check with the owner_id of the map before sending it back once auth is in place.

**Review by kitzbergerg — APPROVED** [review]
Backend looks good (again I haven't looked at the frontend).

**Comment by markus2330** [comment]
jenkins build please

**Comment by markus2330** [comment]
jenkins build please

File: backend/src/controller/map.rs (hunk 3 above)
Comment (markus2330) [reply backend/src/controller/map.rs:47]: Thanks for telling us, can you please fix this in #330 (or create a follow-up issue).

**Review by markus2330 — APPROVED** [review]

**Comment by markus2330** [comment]
@lukashartl there is a problem with the Docker daemon and l1 is disconnected.

```
+ docker build -t permaplant-mdbook ./ci/container-images/permaplant-mdbook
ERROR: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
script returned exit code 1
```

**Pull request merged**